metadata-ingestion/tests/integration/powerbi/test_m_parser.py

```diff
@@ -348,6 +350,61 @@ def test_google_bigquery_2():
     )
 
 
+def test_for_each_expression_1():
+    table: powerbi_data_classes.Table = powerbi_data_classes.Table(
+        expression=M_QUERIES[19],
+        name="D_WH_DATE",
+        full_name="my-test-project.universal.D_WH_DATE",
+    )
+
+    reporter = PowerBiDashboardSourceReport()
+
+    data_platform_tables: List[DataPlatformTable] = parser.get_upstream_tables(
+        table,
+        reporter,
+        native_query_enabled=False,
+        parameters={
+            "Parameter - Source": "my-test-project",
+            "My bq project": "gcp_billing",
+        },
+    )
+
+    assert len(data_platform_tables) == 1
+    assert data_platform_tables[0].name == table.full_name.split(".")[2]
+    assert data_platform_tables[0].full_name == table.full_name
+    assert (
+        data_platform_tables[0].data_platform_pair.powerbi_data_platform_name
+        == SupportedDataPlatform.GOOGLE_BIGQUERY.value.powerbi_data_platform_name
+    )
+
+
+def test_for_each_expression_2():
+    table: powerbi_data_classes.Table = powerbi_data_classes.Table(
+        expression=M_QUERIES[20],
+        name="D_GCP_CUSTOM_LABEL",
+        full_name="my-test-project.gcp_billing.D_GCP_CUSTOM_LABEL",
```

Comment:
in the original query, this `"dwh-prod"` was not a variable reference, so this should not get resolved to `my-test-project`

Reply:
I think a call would be get to understand what is variable reference?

Reply:
No - "dwh-prod" is a string and not a variable reference, and hence should not be resolved. See my comment above as well.

```suggestion
        full_name="dwh-prod.gcp_billing.D_GCP_CUSTOM_LABEL",
```

Reply:
done